### PR TITLE
Author datetime fix

### PR DESCRIPTION
**Pull request by jbradl11**
Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

Update Author datetime functionality for datatypes that can have relevant period and author datetime. Additionally if one of those datatypes are negated continue to use the start time. Updated to latest cql4browsers.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1532
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] ~Automated regression test(s) pass:~ N/A Many documented regressions.

If JIRA tests were used to supplement or replace automated tests:
- [x] ~JIRA test links:~
- [x] ~Justification for using JIRA tests:~
- [x] ~JIRA tests have been added to sprint~

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/cqlpatient.js.coffee [hunk 1]
```diff
@@ -194,7 +194,7 @@ class CQL_QDM.CQLPatient
           # e.g. "Encounter, Performed: Face-to-Face Interaction" becomes
           # EncounterPerformed
           # TODO: this needs to be modified to look at the HQMF template OID instead of the description
-          classname = dc.description.substr(0, dc.description.lastIndexOf(':'))
```
Comment (hossenlopp) [inline app/assets/javascripts/cqlpatient.js.coffee]: This change seems un-related to authorDatetime changes. Is this expected to be here?

Comment (jbradl11) [reply app/assets/javascripts/cqlpatient.js.coffee]: Good catch! This is the fix for: https://oncprojectracking.healthit.gov/support/browse/BONNIE-546

But I forgot to document it in the PR. I'll update the description
